lib/services/cloud-platform-service.ts

```diff
@@ -0,0 +1,120 @@
+import * as semver from "semver";
+import * as constants from "../constants";
+
+export class CloudPlatformService implements ICloudPlatformService {
+	private shouldUseOldPlatformService: boolean;
+
+	constructor(
+		private $injector: IInjector,
+		private $staticConfig: IStaticConfig
+	) {
+		const cliVersion = this.$staticConfig.version;
+		this.shouldUseOldPlatformService = semver.valid(cliVersion) && semver.lt(cliVersion, semver.prerelease(cliVersion) ? "5.4.0-2019-05-16-13277" : "6.0.0");
+	}
+
+	public get $nsCloudBuildCommandHelper(): IBuildCommandHelper {
+		return this.$injector.resolve("nsCloudBuildCommandHelper");
+	}
+
+	public async shouldBuild(config: IApplicationBuildConfig, projectData: IProjectData): Promise<boolean> {
+		if (this.shouldUseOldPlatformService) {
+			const $platformService = this.$injector.resolve("platformService");
+			return $platformService.shouldBuild(config.platform, projectData, config, config.outputPath);
+		}
+
+		const $buildController = this.$injector.resolve("buildController");
+		return $buildController.shouldBuild(config);
+	}
+
+	public async shouldInstall(config: IApplicationInstallConfig, projectData: IProjectData, device: Mobile.IDevice): Promise<boolean> {
+		if (this.shouldUseOldPlatformService) {
+			const $platformService = this.$injector.resolve("platformService");
+			return $platformService.shouldInstall(device, projectData, config, config.outputPath);
+		}
+
+		const $deviceInstallAppService = this.$injector.resolve("deviceInstallAppService");
+		return $deviceInstallAppService.shouldInstall(device, config);
+	}
+
+	public async deployPlatform(platform: string, outputDirectoryPath: string): Promise<void> {
+		const $deployCommandHelper = this.$injector.resolve("deployCommandHelper");
```

Comment:
Why can't we leave the injector to resolve this in the constructor. We seem to use different method, but the same helper.